Communicate to user if db backup files could not be deleted

schasi opened the pull request:
### Description

One user reports that, when enabling automatic database export, files get created but not deleted. This is somewhat  hard to debug without direct access to that users phone.
Also, there is currently no code to check if deletion was actually successfull. Instead it is assumed that it always succeeds when a directory can be written to, which might be wrong for some devices.

This PR adds code to check if backup files could be deleted and show an error message to the user if not. This could be used as a starting point for further investigation.

Supports #7869

### Checklist
<!-- 
  To help us keep the issue tracker clean and work as efficient as possible,
  please make sure that you have done all of the following.
  You can tick the boxes below by placing an x inside the brackets like this: [x]
-->
- [X] I have read the contribution guidelines: https://github.com/AntennaPod/AntennaPod/blob/develop/CONTRIBUTING.md#submit-a-pull-request
- [X] I have performed a self-review of my code
- [X] I have run the automated code checks using `./gradlew checkstyle spotbugsPlayDebug spotbugsDebug :app:lintPlayDebug`
- [X] My code follows the style guidelines of the AntennaPod project: https://github.com/AntennaPod/AntennaPod/wiki/Code-style
- [X] I have mentioned the corresponding issue and the relevant keyword (e.g., "Closes: #xy") in the description (see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [X] If it is a core feature, I have added automated tests

keunes commented:
Thanks @schasi !
I don't know if @ByteHamster would accept this one in, but if so, any content shown to the end-user (including error messages) should be translated. It seems you've currently not foreseen this (?)

While at it, I would also
* Make it plural-ready so we can distinguish an issue with a sing and more than one databases.
* Write out 'dagabase' as not all users might realise what 'db' stands for.

Also, this could be shown at any point, even if the user isn't actively using the app, right? Shouldn't this be shown as a proper notification, which allows the user to follow up? E.g. open the folder, see error message details, send debugging report (depending on what we expect the user to do with this info).

schasi commented:
Thanks for your input @keunes 

The code I added is very similar to already existing code in the same file. There are several `IOException`s thrown, with a reason, which get caught and relayed to the user. The message shown to the user consists of a translated part ("Error during automatic database backup") and the untranslated reason of the `IOException`.

How this gets shown to the user also seems to depend on the state of the app: It could be either a SnackBar, a real Notification or a Toast. So the suggestion you made already seems implemented; I just added another error case to display.

The way I interpret the intention of the code is that the user gets notified of something going wrong instead of it being silent, so he can take that error message and get back to us, as he cannot do something on his own. These errors are of the "the manufacturer messed up the Android system, we developers need to work around it" kind, as I suspect in the linked bug report. So I guess it's ok if the actual error is not translated, as it's more for us than the user?
